model/postgres/postgres_server.rb

```diff
@@ -149,11 +149,12 @@ def needs_recycling?
 
   def lsn_caught_up
     parent_server = if read_replica?
-      resource.parent.representative_server
+      resource.parent&.representative_server
     else
       resource.representative_server
     end
-    lsn_diff(parent_server.current_lsn, current_lsn) < 80 * 1024 * 1024
+
+    lsn_diff(parent_server&.current_lsn || current_lsn, current_lsn) < 80 * 1024 * 1024
```

Comment:
jw: Can condition this be hit in an unfortunate timing case during a genuine failover? For eg. the primary becomes unhealthy and another server starts taking over by removing the primary, and someone creates a read replica around that moment and it get marked as caught up.

I imagine a very low probably case so safe to ignore.